ccp/notifications/notify.py

```diff
@@ -57,17 +67,32 @@ def subject_of_email(self, status, image_name, registry_url):
             return self.build_failure_subj.format(
                 registry=registry, image_name=image_name)
 
-    def body_of_email(self, status, repository, cause):
+    def body_of_email(self, status, image_name, cause,
+                      registry_url, registry_alias):
         """
         Generate the body of email using given details
         :param status: Status of container image build
         :type status bool
-        :param repository: Repository name of image along with registry name
-        :type repository str
+        :param image_name: Container image name without registry
+        :type image_name str
         :param cause: Cause of the the build
         :type cause str
+        :param registry_url: Configured registry URL
+        :type registry_url str
+        :param registry_alias: Registry alias to be used in repository name,
+            if it's value is "null", the $registry_url is used instead
+        :type registry_alias str
         :return: Email body of notification email in text
         """
+        if registry_alias == "null":
+            # get the repository name (without tag)
+            repository = "https://" + registry_url + \
+                "/" + image_name.split(":")[0]
+        else:
+            # if registry_alias is not "null", use it
+            repository = "https://" + registry_alias + \
+                "/" + image_name.split(":")[0]
+
```

Comment:
cant we make the checks in single place to avoid repetition? also if we update it later we will not need to update in both the places.

Reply:
I thought about this, however I kept in all the respective methods who have to deal with these values in order to scope the presentation of these values.
For eg:

`subject_of_email` method has all the values, and its knows how to re-present those values, similarly
`body_of_email` method has all the the  values, and it knows how to formulate those values in email text.

I think this way, we'll have more control over whats formatted for a particular piece in notification email.